egs/tedlium/s5_r2/local/run_segmentation_long_utts.sh

```diff
@@ -0,0 +1,264 @@
+# %WER 17.7 | 1155 27500 | 84.6 11.4 4.1 2.3 17.7 86.6 | -0.018 | exp/tri3_reseg_a_cleaned_c/decode_nosp_test_rescore/score_14_0.0/ctm.filt.filt.sys
+
+## Use silence and pronunciation probs estimated from resegmented data
+# %WER 18.2 | 507 17783 | 84.6 10.8 4.5 2.9 18.2 92.5 | -0.037 | exp/tri3_reseg_a/decode_a_dev_rescore/score_16_0.0/ctm.filt.filt.sys
+# %WER 16.9 | 1155 27500 | 85.5 11.0 3.5 2.4 16.9 86.1 | -0.024 | exp/tri3_reseg_a/decode_a_test_rescore/score_14_0.0/ctm.filt.filt.sys
+
+## Use silence and pronunciation probs estimated from resegmented and cleaned up data
+# %WER 18.2 | 507 17783 | 84.4 10.8 4.9 2.6 18.2 92.5 | -0.074 | exp/tri3_reseg_a_cleaned_b/decode_a_cleaned_b_dev_rescore/score_15_0.5/ctm.filt.filt.sys
+# %WER 16.8 | 1155 27500 | 85.4 10.7 3.9 2.1 16.8 86.8 | -0.046 | exp/tri3_reseg_a_cleaned_b/decode_a_cleaned_b_test_rescore/score_14_0.5/ctm.filt.filt.sys
+
+. ./cmd.sh
+. ./path.sh
+
+set -e -o pipefail -u
+
+segment_stage=-9
+cleanup_stage=-1
+cleanup_affix=cleaned_b
+affix=_a
+
+decode_nj=8   # note: should not be >38 which is the number of speakers in the dev set
+              # after applying --seconds-per-spk-max 180.  We decode with 4 threads, so
+              # this will be too many jobs if you're using run.pl.
+
+###############################################################################
+## Simulate unsegmented data directory.
+###############################################################################
+utils/data/convert_data_dir_to_whole.sh data/train data/train_long
+
+utils/subset_data_dir.sh --speakers data/train 2000 data/train_2k
+utils/subset_data_dir.sh --shortest data/train_2k 500 data/train_2k_short500
+
+steps/make_mfcc.sh --cmd "$train_cmd" --nj 32 \
+  data/train_long exp/make_mfcc/train_long mfcc || exit 1
+steps/compute_cmvn_stats.sh data/train_long \
+  exp/make_mfcc/train_long mfcc
+
+steps/train_mono.sh --nj 20 --cmd "$train_cmd" \
+  data/train_2k_short500 data/lang_nosp exp/mono_a
+
+steps/align_si.sh --nj 20 --cmd "$train_cmd" \
+  data/train_2k data/lang_nosp exp/mono_a exp/mono_a_ali_2k
+
+steps/train_deltas.sh --cmd "$train_cmd" \
+  500 5000 data/train_2k data/lang_nosp exp/mono_a_ali_2k exp/tri1a
+
+steps/align_si.sh --nj 20 --cmd "$train_cmd" \
+  data/train_2k data/lang_nosp exp/tri1a exp/tri1a_ali
+
+steps/train_lda_mllt.sh --cmd "$train_cmd" \
+  1000 10000 data/train_2k data/lang_nosp exp/tri1a_ali exp/tri1b
+
+###############################################################################
+# Segment long recordings using TF-IDF retrieval of reference text 
+# for uniformly segmented audio chunks based on Smith-Waterman alignment.
+# Use a model trained on train_si84 (tri2b)
+###############################################################################
+bash -x steps/cleanup/segment_long_utterances.sh --cmd "$train_cmd" \
+  --stage $segment_stage \
+  --config conf/segment_long_utts.conf \
+  --max-segment-duration 30 --overlap-duration 5 \
```

Comment:
Think carefully about which options to retain here and what the defaults should be.  
I'd suggest only to specify options here if there's some reason it should differ from the
default, or if most users will need to tune it.